src/main/java/de/tum/in/www1/artemis/service/ResourceLoaderService.java

```diff
@@ -34,13 +34,32 @@ public ResourceLoaderService(ResourceLoader resourceLoader) {
         allowedOverridePrefixes.add(Path.of("templates", "jenkins").toString());
     }
 
+    /**
+     * Loads the resource from the specified path.
+     * <p>
+     * Only relative paths are allowed.
+     *
+     * @param path A relative path to a resource.
+     * @return The resource located by the specified path.
+     * @throws IllegalArgumentException Thrown in case an absolute path was given.
+     */
+    public Resource getResource(final Path path) throws IllegalArgumentException {
```

Comment:
_Note_: Proper refactoring of this class from `String` paths to `Path` paths already planned/WIP.
Just added the minimal implementation in this PR to keep the scope small, but already make the new methods usable in the `ProgrammingExerciseRepositoryService` to avoid another round of refactoring there.